homeassistant/components/smappee/config_flow.py

```diff
@@ -21,3 +27,69 @@ class SmappeeFlowHandler(
     def logger(self) -> logging.Logger:
         """Return logger."""
         return logging.getLogger(__name__)
+
+    async def async_step_zeroconf(self, discovery_info=None):
+        """Handle zeroconf discovery."""
+        if discovery_info is None:
+            return self.async_abort(reason="connection_error")
+
+        if not discovery_info[CONF_HOSTNAME].startswith("Smappee1"):
+            # We currently only support Energy and Solar models (legacy)
+            return self.async_abort(reason="invalid mDNS")
```

Comment:
Please use lowercase snakecase string for the abort reason and make sure to add it to strings.json.